l2l/optimizers/kalmanfilter/optimizer.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import numpy as np
+
+from collections import namedtuple
+from l2l.optimizers.kalmanfilter.enkf import EnsembleKalmanFilter as EnKF
+from l2l import dict_to_list
+from l2l.optimizers.optimizer import Optimizer
+from l2l.optimizers.crossentropy.distribution import Gaussian
+
+logger = logging.getLogger("optimizers.kalmanfilter")
+
+EnsembleKalmanFilterParameters = namedtuple(
+    'EnsembleKalmanFilter', ['noise', 'gamma', 'maxit', 'n_iteration',
+                             'pop_size', 'n_batches', 'online', 'seed',
+                             'data', 'observations', 'sampling_generation']
+)
+
+EnsembleKalmanFilterParameters.__doc__ = """
+:param noise: float, Noise level
+:param gamma
+:param maxit: int, Epochs to run inside the Kalman Filter
+:param n_iteration: int, Number of iterations to perform
+:param pop_size: int, Minimal number of individuals per simulation.
+:param n_batches: int, Number of mini-batches to use in the Kalman Filter
+:param online: bool, Indicates if only one data point will used, 
+               Default: False
+:param sampling_generation: After `sampling_gerneration` steps a gaussian sampling 
+        on the parameters of the best individual is done, ranked by the fitness
+        value 
+:param seed: The random seed used to sample and fit the distribution. 
+             Uses a random generator seeded with this seed.
+:param data: nd numpy array, numpy array containing data in format 
+             (batch_size, data)
+:param observations: nd numpy array, observation or targets, should be an
+                     array of integers
+"""
+
```

Comment:
At the moment I think that this optimizer is only applicable to a subset of optimizees. I consider it would be important to document requirements from the optimizee in order to make use of this optimizer, specially in the way the individuals must be formed.